unit-tests/src/main/scala/scala/scalanative/native/CComplexSuite.scala

```diff
@@ -0,0 +1,291 @@
+package scala.scalanative.native
+
+import complex._
+import complexOps._
+
+import java.{lang => jl}
+
+// Reference
+// http://en.cppreference.com/w/c/numeric/complex
+// Ran the complex functions in C to get the results
+// in hex which are used here for the tests.
+// TODO: 2 float tests fail on Linux so they are omitted for now
```

Comment:
They work on macOS and all the other tests work including the double versions of those functions so there has to be some floating point difference on one of the platforms. I tried to add `Math.ulp` to the comparison but it didn't seem to help.